### DISCUSSION TITLE
Is it possible to query or export the Revisions bin?

### DISCUSSION THREAD

**Discussion by RvZEU** [ravendb/ravendb#16523] [Q&A] [answered]
Hi,

we have deleted some documents, but a lot of them we deleted by error.

These documents are deleted by error. Is it possible to query the revisions bin or export it somehow so that we can reimport these items.

Thanx in advance.

**Comment by ayende** [comment, marked as answer]
Only via the REST endpoint, I'm afraid.
`GET /databases/$DB_NAME/revisions/bin`

**Reply by RvZEU** [reply]
Thank you! i see we can also paginate through these items, so it should work.

**Reply by innermark** [reply]
Is there any documentation or examples of this?

**Reply by ppekrol** [reply]
There are two query string parameters there, `start` and `pageSize` to control the pagination.
```
GET /databases/$DB_NAME/revisions/bin?start=0&pageSize=25
```

**Reply by innermark** [reply]
Is this still true that the only way is through the REST endpoint, Claude keeps telling me there is a way to query via RQL now, but I can't get it to work or find an example

**Reply by ppekrol** [reply]
Hi @innermark, nothing changed in that area recently.
